src/lib_ccx/params.c

```diff
@@ -34,47 +34,58 @@
 
 static int inputfile_capacity=0;
 
-int process_cap_file (char *filename)
+#define MAX_WORD_LENGTH 50
+
+size_t remove_trailing_whitespace(char *line)
+{
+	char *c = line + strlen(line) - 1;
+	while (c >= line && isspace(*c))
+		c--;
+	c[1] = '\0';
+	return c - line;
+}
+
+int process_word_file(const char *filename, int (*add_word_to_list)(const char *))
 {
 	int ret = 0;
-	FILE *fi = fopen (filename,"rt");
-	if (fi == NULL)
+	FILE *fi;
+
+	if ((fi = fopen(filename, "r")) == NULL)
 	{
-		mprint ("\rUnable to open capitalization file: %s\n", filename);
+		mprint("Unable to open word list file: %s\n", filename);
```

Comment:
You're missing \r there, which is there so the message appears at the beginning of the line (instead of to the right of the progress info)

Reply:
What's the difference between \r and \n?

Should a '\n' be added instead?

Reply:
No. \r moves the cursor to the beginning of the line and \n starts a new line. So if you want to be in the same line would start writing from the left (overwriting whatever is already there) you send a \r.

Reply:
I had no clue this was possible. I will probably use this a lot now.

Reply:
That is basically what CRLF is, Carriage Return (\r) and Line Feed (\n)